apps/condo/domains/miniapp/schema/B2BAccessToken.js

```diff
@@ -0,0 +1,276 @@
+            knexOptions: { isNotNullable: true }, // Required relationship only!
+            kmigratorOptions: { null: false, on_delete: 'models.CASCADE' },
+            access: access.updatableOnlyForAdmin,
+        },
+
+        user: {
+            ...SERVICE_USER_FIELD,
+            isRequired: true,
+            access: access.updatableOnlyForAdmin,
+            hooks: {
+                ...SERVICE_USER_FIELD.hooks,
+                async resolveInput ({ operation, existingItem, resolvedData, fieldPath }) {
+                    const newItem = {
+                        ...existingItem,
+                        ...resolvedData,
+                    }
+                    if (operation === 'create' && !newItem[fieldPath]) {
+                        const contextId = get(resolvedData, 'context') || get(existingItem, 'context')
+                        const context = await getById('B2BAppContext', contextId)
+                        const [accessRight] = await find('B2BAppAccessRight', {
+                            app: { id: context.app },
+                            deletedAt: null,
+                        })
+                        return accessRight.user
+                    }
+
+                    return resolvedData.user
+                },
+            },
+        },
+
+        rightSet: {
+            schemaDoc: 'Right set for token',
+            type: 'Relationship',
+            ref: 'B2BAppAccessRightSet',
+            isRequired: true,
+            knexOptions: { isNotNullable: true }, // Required relationship only!
+            kmigratorOptions: { null: false, on_delete: 'models.CASCADE' },
+            access: access.updatableOnlyForAdmin,
+        },
+
+        expiresAt: {
+            schemaDoc: 'Date, when access by this token will be restricted',
+            type: 'DateTimeUtc',
+            isRequired: false,
+            access: access.updatableOnlyForAdmin,
+            hooks: {
+                validateInput ({ resolvedData, fieldPath, context }) {
+                    const expiresAt = resolvedData[fieldPath]
+                    if (isNil(expiresAt)) {
+                        return
+                    }
+                    if (dayjs(expiresAt).isBefore(dayjs())) {
+                        throw new GQLError(ERRORS.NEGATIVE_TTL, context)
+                    }
+                },
+            },
+        },
+    },
+    hooks: {
+        async validateInput ({ operation, resolvedData, existingItem, context, originalInput }) {
+            const newItem = {
+                ...existingItem,
+                ...resolvedData,
+            }
+
+            if (isSoftDelete(originalInput)) {
+                return
+            }
+
+            const validationError = await getValidationError(newItem)
+            if (operation === 'create' && validationError) {
+                throw new GQLError(validationError, context)
+            }
+            if (operation === 'update' && validationError) {
```

Comment:
Can you explain this branch?

What is ERROR_FOR_DELETION, what's happening with deletedAt? Seems like some not obvious hacks